### PR TITLE
fix(auth): add extra notification to SSO login flow

### PR DESCRIPTION
**Pull request by JadenSimon**
## Problem
Logging in via the browser is currently too easy. [Section 5.4 of RFC 8628](https://www.rfc-editor.org/rfc/rfc8628#section-5.4) suggests that implementations should make it clear to users _which_ device is requesting access at all times. 

## Solution
Add an additional notification containing the user verification code before opening the browser. Users must now enter in this code in the browser whereas before it was done automatically. The user code is automatically copied to the clipboard to make things slightly easier.

### Screenshots
**The help button won't be present in the initial release**

Screen Shot 2023-02-02 at 4 27 46 PM

Screen Shot 2023-02-03 at 8 13 43 AM

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/constants.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export const documentationUrl: string = isCloud9()
  * - alternative?: codecatalyst/latest/userguide/sign-up-create-resources.html
  */
 export const authHelpUrl = 'https://docs.aws.amazon.com/general/latest/gr/differences-aws_builder_id.html'
+export const ssoAuthHelpUrl = 'https://docs.aws.amazon.com/singlesignon/latest/userguide/howtosignin.html'
```
Comment (JadenSimon) [inline src/shared/constants.ts:31]: This was unfortunately the best I could find. There apparently isn't a generic documentation page for the device code flow. The only thing I found was for the AWS CLI which would obviously be confusing too.

File: .changes/next-release/Bug Fix-2564ea73-0e4a-4404-ab79-14710c867bea.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Bug Fix",
+	"description": "Authenticating through the browser now requires users to manually enter a user verification code. Previously this was done automatically."
```
Comment (justinmk3) [inline .changes/next-release/Bug Fix-2564ea73-0e4a-4404-ab79-14710c867bea.json:3]: should we mention "AWS Builder ID" here

**Review by justinmk3 — APPROVED** [review]

File: src/test/credentials/sso/model.test.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+import * as assert from 'assert'
+import * as sinon from 'sinon'
+import * as vscode from 'vscode'
+import { createTestWindow } from '../../shared/vscode/window'
+import { openSsoPortalLink } from '../../../credentials/sso/model'
+import { assertTelemetry } from '../../testUtil'
+import { CancellationError } from '../../../shared/utilities/timeoutUtils'
+
+describe('openSsoPortalLink', function () {
```
Comment (JadenSimon) [inline src/test/credentials/sso/model.test.ts:14]: Normally I wouldn't suggest writing thorough tests for such a small component but it's good to have more examples of one way to test UI.

File: src/test/credentials/sso/model.test.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    let testWindow: ReturnType<typeof createTestWindow>
+    let openExternal: sinon.SinonStub<any[], Thenable<boolean>>
+
+    beforeEach(function () {
+        sinon.replace(vscode, 'window', (testWindow = createTestWindow()))
+        openExternal = sinon.stub(vscode.env, 'openExternal').resolves(true)
```
Comment (JadenSimon) [inline src/test/credentials/sso/model.test.ts:20]: We can probably de-dupe this into a test util.

File: .changes/next-release/Bug Fix-2564ea73-0e4a-4404-ab79-14710c867bea.json (hunk 2 above)
Comment (JadenSimon) [reply .changes/next-release/Bug Fix-2564ea73-0e4a-4404-ab79-14710c867bea.json:3]: It applies to all SSO logins but I suppose we could add specific examples somewhere

**Review by gsweetwood — APPROVED** [review]

File: src/shared/constants.ts (hunk 1 above)
Comment (justinmk3) [reply src/shared/constants.ts:31]: The best reference that I found is the one just above this. 

Some ongoing discussion about potentially a new page.

**Review by justinmk3 — APPROVED** [review]

File: src/test/credentials/sso/model.test.ts (hunk 4 above)
Comment (JadenSimon) [reply src/test/credentials/sso/model.test.ts]: This is now done globally in https://github.com/aws/aws-toolkit-vscode/pull/3148/commits/e2d73035e521f290aa9af56abc391448cc369b12

**Pull request merged**